ui/raidboss/emulator/data/PopupTextAnalysis.ts

```diff
@@ -81,9 +81,14 @@ export class Resolver {
   }
 }
 
+export type LineRegExpCache = Map<LineEvent, Map<ProcessedTrigger, RegExpExecArray | false>>;
```

Comment:
All the line matching seems very reasonable to me!

One idle thought I was having (and I'm not sure how complicated this would look like) but would be to do a first pass where you match all lines against all (trigger regex / timeline regex) during log processing.  My assumption is that most lines won't match and so then you'd have 10% (vast overestimate) lines to store in idb.  I don't know how you'd detect if somebody added more triggers or changed regexes though, and so maybe this isn't a plausible approach.  I guess you could do it once after loading here but that's not quite the same level of savings.

Reply:
By "During log processing" do you mean during the initial log import? Because if that's the case, I don't really think it would make sense in general. That would break combatant tracking, which would break getCombatants triggers.

Reply:
Mmm, that's fair.  I always forget how many lines are needed for that.